services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/llm_prompt.rb

```diff
@@ -64,7 +64,8 @@ def text = prompt
         def self.create_from_template!(dataset_id:, guideline_ids:, llm_prompt_template_id:, prompt_example_ids:, relevant_text_id:)
           guidelines = Guideline.where(id: guideline_ids)
           prompt_examples = PromptExample.where(id: prompt_example_ids)
-          relevant_text = RelevantText.find(relevant_text_id)
+          relevant_text = RelevantText.find(relevant_text_id) if relevant_text_id.present?
```

Comment:
Take or leave, since this is purely stylistic, but I tend to prefer using `.find_by(id: relevant_text_id)` over a guard clause as that will return `nil` if there is no match (as would be the case if `relevant_text_id` is `nil`).  Though that would open up the possibility of a present `relevant_text_id` not finding a match and failing to error, so your current approach may be safer.

Reply:
I actually like your suggestion better here as I don't actually want this to error.